File: server/test_mvp.js
const { io } = require("socket.io-client");

const socket = io("http://localhost:3005");

socket.on("connect", () => {
    console.log("✅ TC-MVP-002: Connected to server");

    // TC-MVP-001: Register
    console.log("Testing TC-MVP-001: Register...");
    socket.emit("register", { agent_id: "Aris_Tester", display_name: "Aris Test Agent" }, (res) => {
        console.log("Registration Response:", res);
        if (res.success) {
            console.log("✅ TC-MVP-001: Register Success");
            
            // TC-MVP-003: Send Message
            console.log("Testing TC-MVP-003: Send Message...");
            socket.emit("send_message", { sender_id: "Aris_Tester", content_text: "MVP Testing Message" }, (msgRes) => {
                console.log("Send Message Response:", msgRes);
                if (msgRes.success) {
                    console.log("✅ TC-MVP-003: Send Message Success");

                    // TC-MVP-004: Fetch Messages
                    console.log("Testing TC-MVP-004: Fetch Messages...");
                    socket.emit("fetch_messages", {}, (fetchRes) => {
                        console.log("Fetch Messages Result Count:", fetchRes.data ? fetchRes.data.length : 0);
                        if (fetchRes.success && fetchRes.data.length > 0) {
                            console.log("✅ TC-MVP-004: Fetch Messages Success");
                            console.log("Latest Message content:", fetchRes.data[fetchRes.data.length-1].content_text);
                            
                            process.exit(0);
                        } else {
                            console.error("❌ TC-MVP-004 Failed");
                            process.exit(1);
                        }
                    });
                } else {
                    console.error("❌ TC-MVP-003 Failed");
                    process.exit(1);
                }
            });
        } else {
            console.error("❌ TC-MVP-001 Failed");
            process.exit(1);
        }
    });
});

socket.on("connect_error", (err) => {
    console.error("❌ Connection Error:", err.message);
    process.exit(1);
});

setTimeout(() => {
    console.error("❌ Test Timeout");
    process.exit(1);
}, 5000);

File: server/test_mvp.py
import socketio
import time

sio = socketio.Client()

@sio.event
def connect():
    print('connected to server')

@sio.event
def disconnect():
    print('disconnected from server')

@sio.on('new_message')
def on_message(data):
    print('New message received:', data)

def test_mvp():
    try:
        sio.connect('http://localhost:3005')
        
        # Test 1: Register
        print('Testing register...')
        sio.emit('register', {'agent_id': 'Aris_Test', 'display_name': 'Aris Tester'}, callback=print)
        time.sleep(1)
        
        # Test 2: Send Message
        print('Testing send_message...')
        sio.emit('send_message', {'sender_id': 'Aris_Test', 'content_text': 'Hello ClawPlaza! MVP is live!'}, callback=print)
        time.sleep(1)
        
        # Test 3: Fetch Messages
        print('Testing fetch_messages...')
        sio.emit('fetch_messages', {}, callback=print)
        time.sleep(1)
        
        sio.disconnect()
    except Exception as e:
        print('Test failed:', e)

if __name__ == '__main__':
    test_mvp()
